packages/plugin-core/src/utils/analytics.ts

```diff
@@ -133,6 +137,91 @@ export class AnalyticsUtils {
     );
   }
 
+  /** Saves analytics to be sent during the next run of Dendron.
+   *
+   * Make sure any properties you use can be trivially serialized and
+   * deserialized, numbers, strings, plain JSON objects, arrays are fine. No
+   * Maps or complex objects.
+   *
+   * This is required for actions that reload the window, where the analytics
+   * won't get sent in time before the reload and where delaying the reload
+   * would be undesirable.
+   */
+  static async trackForNextRun(event: string, customProps?: any) {
+    const ctx = "AnalyticsUtils.trackForNextRun";
+    Logger.debug({
+      ctx,
+      event,
+    });
+    const analyticsProps = this._trackCommon({
+      event,
+      props: {
+        ...customProps,
+        savedAnalytics: true,
+      },
+      timestamp: new Date(),
+    });
+    const telemetryDir = path.join(
+      os.homedir(),
+      FOLDERS.DOTFILES,
+      FOLDERS.SAVED_TELEMETRY
+    );
+    await fs.ensureDir(telemetryDir);
+    await fs.writeFile(
+      path.join(telemetryDir, `${genUUID()}.json`),
+      JSON.stringify(analyticsProps)
+    );
+  }
+
+  static async sendSavedAnalytics() {
+    const ctx = "AnalyticsUtils.sendSavedAnalytics";
+    Logger.debug({ ctx, message: "start" });
+    const telemetryDir = path.join(
+      os.homedir(),
+      FOLDERS.DOTFILES,
+      FOLDERS.SAVED_TELEMETRY
+    );
+    let files: string[] = [];
+    try {
+      files = await fs.readdir(telemetryDir);
+    } catch {
+      Logger.warn({
+        ctx,
+        msg: "failed to read the saved telemetry dir",
+        telemetryDir,
+      });
+    }
+
+    return asyncLoop(
+      files.filter((filename) => path.extname(filename) === ".json"),
+      async (filename) => {
+        const filePath = path.join(telemetryDir, filename);
+        try {
+          const contents = await fs.readFile(filePath, { encoding: "utf-8" });
+          const payload = JSON.parse(contents, (_key, value) => {
+            // If we have a timestamp, that will get serialized as a string. We have to deserialize it here.
+            if (
+              _.isString(value) &&
+              // yyyy-mm-ddThh:mm:ss.msZ
+              value.match(/\d\d\d\d-\d\d-\d\dT\d\d:\d\d:\d\d[.]\d+Z/)
+            ) {
+              return new Date(value);
+            }
+            return value;
+          });
+          await fs.rm(filePath);
+          return SegmentUtils.trackSync(payload);
```

Comment:
This was deliberate to avoid double-sending the telemetry in case something fails between us sending the telemetry and the file being deleted. I thought that missing an event is "less bad" than duplicating events.

Does segment/amplitude have a way to de-duplicate events if the timestamps match? If that's the case this is not a concern of course, I can rearrange them.

Reply:
got it. lets keep this for now. we could imagine something more elaborate (eg. add our own transaction layer, write the intend to commit, track sync, after track sync, write commit has been successful, then remove) but not worth it at this point

Reply:
We can probably trigger the behavior off the result of the promise returned from track.  If promise fails, means that telem most likely did not make it, in which case we don't want to delete the data store.  If it succeeds, it means it is guaranteed to have made it up, so we can delete. 

Re: duplicated data vs no data- I think having duplicated data is better than no data? Not sure.

Reply:
But the workspace might close after we sent the telemetry, but before the file is deleted. Then we'll still end up sending duplicate events. I think missing an event is safer because it gives us a more conservative view of what's happening, to make sure we don't over-estimate the usage of certain features.

In any case, this would be a pretty rare case and one missing or duplicates telemetry is not a dealbreaker. I'm happy to modify the code if you insist.

Reply:
So far from looking at data, I'm not as worried about the overcounting problem - we mostly use the 'unique' users view , which wouldn't be affected by this kind of issue. 

We have spent a large amount of time trying to debug ghost issues though - for example "How did this user go from A -> C, but skipped B? Do we have some bug in our code?" But it turns out it was just an instrumentation issue.  For this reason, I'd rather err on over-reporting than under-reporting for now.  @kevinslin what do you think?

Reply:
Got it, I flipped the order around. You're right that we usually rely on unique users views, and all the other precautions are likely enough to avoid duplicate events in real world cases.